src/ucs/sys/sock.c

```diff
@@ -100,22 +100,78 @@ ucs_status_t ucs_netif_ioctl(const char *if_name, unsigned long request,
     return status;
 }
 
-int ucs_netif_is_active(const char *if_name)
+int ucs_netif_get_addr(const char *if_name, uint8_t sa_family,
+                       struct sockaddr *saddr, struct sockaddr *netmask)
 {
-    ucs_status_t status;
-    struct ifreq ifr;
+    ucs_status_t status = UCS_ERR_NO_DEVICE;
+    struct ifaddrs *ifa;
+    struct ifaddrs *ifaddrs;
+    const struct sockaddr_in6 *saddr6;
+    size_t addrlen;
 
-    status = ucs_netif_ioctl(if_name, SIOCGIFADDR, &ifr);
-    if (status != UCS_OK) {
-        return 0;
+    if (getifaddrs(&ifaddrs)) {
+        ucs_warn("getifaddrs error: %m");
+        status = UCS_ERR_IO_ERROR;
+        goto out;
+    }
+
+    for (ifa = ifaddrs; ifa != NULL; ifa = ifa->ifa_next) {
+        if ((if_name != NULL) && (0 != strcmp(if_name, ifa->ifa_name))) {
+            continue;
+        }
+
+        if ((ifa->ifa_addr == NULL) ||
+            ((ifa->ifa_addr->sa_family != AF_INET) &&
+             (ifa->ifa_addr->sa_family != AF_INET6))) {
+            continue;
+        }
+
+        if (!ucs_netif_flags_is_active(ifa->ifa_flags)) {
+            continue;
+        }
+
+        if (ifa->ifa_addr->sa_family == AF_INET6) {
+            saddr6 = (const struct sockaddr_in6*)ifa->ifa_addr;
+            if (IN6_IS_ADDR_LOOPBACK(&saddr6->sin6_addr) ||
+                IN6_IS_ADDR_LINKLOCAL(&saddr6->sin6_addr)) {
+                continue;
+            }
+        }
+
+        if (sa_family == AF_UNSPEC || ifa->ifa_addr->sa_family == sa_family) {
+            status = ucs_sockaddr_sizeof(ifa->ifa_addr, &addrlen);
+            if (status != UCS_OK) {
+                goto out_free_ifaddr;
+            }
+
+            if (saddr != NULL) {
+                memcpy(saddr, ifa->ifa_addr, addrlen);
+            }
+
+            if (netmask != NULL) {
+                memcpy(netmask, ifa->ifa_netmask, addrlen);
+            }
+
+            status = UCS_OK;
+            break;
+        }
     }
+out_free_ifaddr:
+    freeifaddrs(ifaddrs);
+out:
+    return status;
+}
 
-    status = ucs_netif_ioctl(if_name, SIOCGIFFLAGS, &ifr);
+int ucs_netif_is_active(const char *if_name)
```

Comment:
maybe check it's active according to the selected address-family?
since we check `ucs_netif_flags_is_active(ifa->ifa_flags)` which is per address entry

Reply:
yes, we probably need to move this config to md. 
otherwise, we would fail to open an interface that does not have the configured address - instead of ignoring it in query_tl_resources.

Reply:
ucs_netif_is_active() ignore the interfaces which do not have configured addresses and are not active.  do you see other reasons to config to md?

Reply:
what if one interface has only ipv4, but iface is configured to use ipv6?
and another interface has ipv6?

Reply:
fixed. moved config to md